### PR TITLE
feat: Add cacert:export command to get self-signed Che CA certificate

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Adds a new command `cacert:export` which retrieves self-signed Che CA certificate and saves it to specified file.
Note, it works only for self-signed certificates.

Tested on `minikube`, `crc` and `openshift`.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16278

### CODE REVIEW

File: src/commands/server/certificate.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    destination: string({
+      char: 'd',
+      description: `Destination where to store Che certificate.
+                    If the destination is a file (might not exist), then the certificate will be saved there in PEM format.
+                    If the destination is a directory, then ${DEFAULT_CA_CERT_FILE_NAME} file will be created there with Che certificate in PEM format.
+                    If this option is ommited, then Che certificate will be stored in user's home directory as ${DEFAULT_CA_CERT_FILE_NAME}`,
+      env: 'CHE_CA_CERT_LOCATION',
+      default: '~'
+    }),
+    'make-path': boolean({
```
Comment (tolusha) [inline src/commands/server/certificate.ts:45]: Not sure it is necessary at all

Comment (mmorhun) [reply src/commands/server/certificate.ts:45]: It might be useful in some cases. Anyway, it is implemented and works fine, see no point in removing...

Comment (AndrienkoAleksandr) [reply src/commands/server/certificate.ts:45]: Maybe we should make path by default without extra flags?

File: src/api/che.ts [hunk 2]
```diff
@@ -107,6 +108,22 @@ export class CheHelper {
...
+    const cheCaSecret = await this.kube.getSecret(CHE_ROOT_CA_SECRET_NAME, cheNamespace)
+    if (!cheCaSecret) {
+      throw new Error('Local Che CA self-signed certificate not found. Are you using self-signed certificate?')
+    }
+
+    if (cheCaSecret.data && cheCaSecret.data['ca.crt']) {
+      return Buffer.from(cheCaSecret.data['ca.crt'], 'base64').toString('ascii')
+    }
+
+    throw new Error(`Secret "${CHE_ROOT_CA_SECRET_NAME}" has invalid format.`)
```
Comment (tolusha) [inline src/api/che.ts:124]: Add pls.
`ca.crt key not found.`

File: src/commands/cacert/export.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      options: ['minikube', 'minishift', 'k8s', 'openshift', 'microk8s', 'docker-desktop', 'crc'],
+    }),
+    destination: string({
+      char: 'd',
+      description: `Destination where to store Che certificate.
+                    If the destination is a file (might not exist), then the certificate will be saved there in PEM format.
+                    If the destination is a directory, then ${DEFAULT_CA_CERT_FILE_NAME} file will be created there with Che certificate in PEM format.
+                    If this option is ommited, then Che certificate will be stored in user's home directory as ${DEFAULT_CA_CERT_FILE_NAME}`,
+      env: 'CHE_CA_CERT_LOCATION',
+      default: '~'
```
Comment (tolusha) [inline src/commands/cacert/export.ts:44]: That is unix specific.
Let's have empty default.

File: src/commands/cacert/export.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    } catch (error) {
+      this.error(error)
+    }
+  }
+
+  /**
+   * Handles certificate target location and returns string which points to the target file.
+   */
+  private prepareTarget(destinaton: string): string {
+    if (destinaton === '~') {
```
Comment (tolusha) [inline src/commands/cacert/export.ts:77]: if (!destination)

File: src/commands/cacert/export.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      return path.join(os.homedir(), DEFAULT_CA_CERT_FILE_NAME)
+    }
+
+    if (fs.existsSync(destinaton)) {
+      return fs.lstatSync(destinaton).isDirectory() ? path.join(destinaton, DEFAULT_CA_CERT_FILE_NAME) : destinaton
+    }
+
+    throw new Error('Given certificate path doesn\'t exist.')
```
Comment (tolusha) [inline src/commands/cacert/export.ts:85]: Which path? Add path into the error

File: src/commands/cacert/export.ts (hunk 3 above)
Comment (mmorhun) [reply src/commands/cacert/export.ts:44]: At the end it isn't. But agree, to avoid confusion.

File: src/commands/cacert/export.ts (hunk 4 above)
Comment (mmorhun) [reply src/commands/cacert/export.ts:77]: Doesn't work with `~` =)

File: src/commands/cacert/export.ts (hunk 5 above)
Comment (mmorhun) [reply src/commands/cacert/export.ts:85]: Make sense

File: src/commands/cacert/export.ts (hunk 4 above)
Comment (tolusha) [reply src/commands/cacert/export.ts:77]: if (!destination) => home dir by default
else path.resolve()

**Pull request merged**